DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/mmhost/README.md

```diff
@@ -0,0 +1,372 @@
+    }
+  }
+};
+```
+![](assets/visio-naive.svg)
+
+### Using `annotated_arg` to configure Avalon memory-mapped host interfaces
+
+The default behaviour of a pointer argument can be overridden by declaring an `annotated_arg` kernel argument.
+
+#### Example 2: A kernel with a single customized Avalon memory-mapped host interface
+(Code can be found under `part2_single_host`).
+```c++
+struct SingleMMIP {
+  // This kernel has 3 annotated pointers, but since they have no properties
+  // specified, this kernel will result in the same IP component as Example 1.
+  sycl::ext::oneapi::experimental::annotated_arg<int *> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *> z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+```
+
+The following table describes the properties under `sycl::ext::intel::experimental` that can be used to customize how the pointer argument is passed to the component. Only one may be specified at a time. 
+
+| Parameter                 | Description
+|---                        |---
+| `register_map`            | Pass the pointer for this memory-mapped host interface through the IP component's control/status register
+| `conduit`                 | Pass the pointer for this memory-mapped host interface through a conduit interface 
+
+
+The following parameters found under `sycl::ext::intel::experimental` and `sycl::ext::oneapi::experimental` can be used to configure an IP component's Avalon memory-mapped host interfaces:
+
+| Parameter                | Default Value | Description
+|---                       |---            |---
+|  `buffer_location<id>`   | N/A           | The address space of the interface that associates with the host. Each unique buffer location will result in a unique Avalon memory-mapped host interface. When `buffer_location` is not specified, then corresponding LSUs will connect to all of the IP's Avalon memory-mapped host interfaces.
+| `awidth<width>`          | 41            | Width of the Avalon memory-mapped host address bus in bits. 
+| `dwidth<width>`          | 64            | Width of the Avalon memory-mapped host data bus in bits. 
+| `latency<value>`         | 1             | Read/Write latency for downstream fixed-latency Avalon memory-mapped agent. For a variable-latency Avalon memory-mapped agent, set `latency<0>`. 
+| `read_write_mode<mode>`  | `read_write`  | Port direction of the interface. (`read_write`, `read` or `write`) 
+| `maxburst<value>`        | 1             | Maximum number of data transfers that can associate with a read or write request. 
+| `alignment<alignment>`   | 1          | Alignment of the Avalon memory-mapped host interface
+| `stable`                 | N/A | User guarantee that the pointer will not change between pipelined invocations of the kernel. The compiler uses this to further optimize the kernel.
```

Comment:
```suggestion
```
This belongs in the other table, since it applies to how the argument is passed to the kernel